Workspace/src/ik/src/Roboarm.cpp

```diff
@@ -50,20 +55,36 @@ void Roboarm::calc_velocities(float endEffectorVX, float endEffectorVY, float en
     + (l1 * l2 * (c1 * s12 - s1 * c12) + l1 * l3 * (c1 * s123 - s1 * c123)) * eP;
 }
 
+
 void Roboarm::calc_pose(bool visualize){
     //calculate the anlges at the next time step
     float nextAngle1 = angleLink1 + velocityLink1 * alpha;
     float nextAngle2 = angleLink2 + velocityLink2 * alpha;
     float nextAngle3 = angleLink3 + velocityLink3 * alpha;
 
-    eeX =
+    poseX[0] =
               lengthLink1 * cos(nextAngle1)
             + lengthLink2 * (cos(nextAngle1) + cos(nextAngle2))
             + lengthLink3 * (cos(nextAngle1) + cos(nextAngle2) + cos(nextAngle3));
-    eeY =
+    poseX[1] =
```

Comment:
if you flip how you store these, you can flatten the tree that's happening. something like:
```
pose[0].x = 0.0;
pose[1]x = pose[0].x + ...
pose[2].x = pose[1].x + ...
...
...
...
```

would be easier to read and less computation